src/App.tsx

```diff
@@ -37,7 +37,7 @@ const DashboardSidebar = () => {
   const appEngineHealth = useFetch(astarte.client.getAppengineHealth);
   const realmManagementHealth = useFetch(astarte.client.getRealmManagementHealth);
   const pairingHealth = useFetch(astarte.client.getPairingHealth);
-  const flowHealth = config.features.flow ? useFetch(astarte.client.getFlowHealth) : null;
+  const flowHealth = useFetch(astarte.client.getFlowHealth);
```

Comment:
useFetch fetches the data immediatly, if flow is disabled we shouldn't query it

Reply:
It's not an issue since we would just discard the error caught by `useFetch`, but conceptually I have the same concern.
I wouldn't keep the line as before either, since hooks must not be used inside a conditional.
I'll rework this code another way

Reply:
you'll make the backend log unnecessary errors. Then we must add a flag to say if we should launch the query immediately or not
```js
  const pairingHealth = useFetch(astarte.client.getPairingHealth);
  const flowHealth = useFetch(astarte.client.getFlowHealth, config.features.flow)
```

```js
const useFetch = (...params, initialFetch = true) => {
  [...]

  useEffect(() => {
    isReady.current = true;
    if (initialFetch) {
      getData(...initialFetchParams);
    }
    return () => {
      isReady.current = false;
    };
  }, []);

}
```

Reply:
Thanks for hint! However reworking `useFetch` params is not straightforward and disabling the initial fetch would also mean adding another `initial` state instead of `loading`.

I tried a different solution and eventually reduced the lines of code too: passing a single function to `useFetch` that checks all relevant APIs at once. This also means less React re-renders.